prototyping/xarray/xdose.py

```diff
@@ -0,0 +1,198 @@
+    pixel array in the specified coordinate system.
+
+    For DICOM RT Dose datasets, these are the x, y, z axes of the
+    dose grid.
+
+    Parameters
+    ----------
+    ds : pydicom.dataset.Dataset
+        A DICOM dataset that contains pixel data. Supported modalities
+        include 'CT' and 'RTDOSE'.
+
+    coord_system : str, optional
+        The coordinate system in which to return the `x`, `y` and `z`
+        coordinates of the DICOM dataset. The accepted, case-insensitive
+        values of `coord_system` are:
+
+        'DICOM' or 'd':
+            Return axes in the DICOM coordinate system.
+
+        'patient', 'IEC patient' or 'p':
+            Return axes in the IEC patient coordinate system.
+
+        'support', 'IEC support' or 's':
+            Return axes in the IEC support coordinate system.
+
+    Returns
+    -------
+    (x, y, z)
+        A tuple containing three `numpy.ndarray`s corresponding to the
+        `x`, `y` and `z` coordinates of the DICOM dataset's pixel array
+        in the specified coordinate system.
+
+    Notes
+    -----
+    Supported scan orientations [1]_:
+
+    =========================== ==========================
+    Orientation                 ds.ImageOrientationPatient
+    =========================== ==========================
+    Feet First Decubitus Left   [0, 1, 0, 1, 0, 0]
+    Feet First Decubitus Right  [0, -1, 0, -1, 0, 0]
+    Feet First Prone            [1, 0, 0, 0, -1, 0]
+    Feet First Supine           [-1, 0, 0, 0, 1, 0]
+    Head First Decubitus Left   [0, -1, 0, 1, 0, 0]
+    Head First Decubitus Right  [0, 1, 0, -1, 0, 0]
+    Head First Prone            [-1, 0, 0, 0, -1, 0]
+    Head First Supine           [1, 0, 0, 0, 1, 0]
+    =========================== ==========================
+
+    References
+    ----------
+    .. [1] O. McNoleg, "Generalized coordinate transformations for Monte
+       Carlo (DOSXYZnrc and VMC++) verifications of DICOM compatible
+       radiotherapy treatment plans", arXiv:1406.0014, Table 1,
+       https://arxiv.org/ftp/arxiv/papers/1406/1406.0014.pdf
+    """
+
+    position = np.array(ds.ImagePositionPatient)
+    orientation = np.array(ds.ImageOrientationPatient)
+
+    if not (
+        np.array_equal(np.abs(orientation), np.array([1, 0, 0, 0, 1, 0]))
+        or np.array_equal(np.abs(orientation), np.array([0, 1, 0, 1, 0, 0]))
+    ):
+        raise ValueError(
+            "Dose grid orientation is not supported. Dose "
+            "grid slices must be aligned along the "
+            "superoinferior axis of patient."
+        )
+
+    is_decubitus = dataset_orientation_is_decubitus(ds)
+    is_head_first = dataset_orientation_is_head_first(ds)
+
+    di = float(ds.PixelSpacing[0])
+    dj = float(ds.PixelSpacing[1])
+
+    col_range = np.arange(0, ds.Columns * di, di)
+    row_range = np.arange(0, ds.Rows * dj, dj)
+
+    if is_decubitus:
+        x_support = orientation[1] * position[1] + col_range  # type: ignore
```

Comment:
Is there a way around this type ignore here? Maybe using type casting if need be?

Reply:
This doesn't complain now...